test/resource_owner_password_grant_test.dart

```diff
@@ -46,6 +46,29 @@ void main() {
       expect(client.credentials.accessToken, equals('2YotnFZFEjr1zCsicMWpAA'));
     });
 
+    test('passes the onCredentialsRefreshed callback to the client', () async {
+      expectClient.expectRequest((request) async {
+        return new http.Response(success, 200,
+            headers: {'content-type': 'application/json'});
+      });
+
+      var onCredentialsRefreshedCallback = (oauth2.Credentials credentials) {};
+
+      var client = await oauth2.resourceOwnerPasswordGrant(
+          authEndpoint, 'username', 'userpass',
+          identifier: 'client',
+          secret: 'secret',
+          httpClient: expectClient,
+          onCredentialsRefreshed: onCredentialsRefreshedCallback);
+
+      var reflection = reflect(client);
```

Comment:
Can't we make assertion that `onCredentialsRefreshedCallback` is called?

Reply:
`onCredentialsRefreshedCallback` is called only when the credentials are refreshed. It won't be invoked when a new access token is granted. So in order to make sure `onCredentialsRefreshedCallback` is indeed called without using `dart:mirror`, I probably need to create a access token response that has a negative `expires_in`, then invoke a request (like what's done in `client_test.dart`) to trigger a token refresh. Not sure if this is a good idea tough..

Reply:
Either that, or we just simply test that providing the parameter doesn't cause a crash :)

But if we were to test that refresh worked after creating a client with `resourceOwnerPasswordGrant` that would be nice, as it would create some tests we don't already have.

But I'm okay, just creating it and checking that specifying the parameter doesn't cause an exception, it's a trivial test, but then we have some testing.